src/backend/marsha/core/tests/api/video/test_upload_ended.py

```diff
@@ -0,0 +1,193 @@
+        playlist_access = factories.PlaylistAccessFactory(
+            playlist=self.some_video.playlist,
+            role=models.STUDENT,
+        )
+
+        self.assert_user_cannot_end_an_upload(playlist_access.user, self.some_video)
+
+    def test_end_upload_by_playlist_instructor(self):
+        """Playlist instructor cannot end an upload."""
+        playlist_access = factories.PlaylistAccessFactory(
+            playlist=self.some_video.playlist,
+            role=models.INSTRUCTOR,
+        )
+
+        self.assert_user_can_end_an_upload(playlist_access.user, self.some_video)
+
+    def test_end_upload_by_playlist_admin(self):
+        """Playlist administrator can end an upload."""
+        playlist_access = factories.PlaylistAccessFactory(
+            playlist=self.some_video.playlist,
+            role=models.ADMINISTRATOR,
+        )
+
+        self.assert_user_can_end_an_upload(playlist_access.user, self.some_video)
+
+    def test_end_upload_with_wrong_body(self):
+        """Playlist administrator can end an upload."""
+        playlist_access = factories.PlaylistAccessFactory(
+            playlist=self.some_video.playlist,
+            role=models.ADMINISTRATOR,
+        )
+
+        jwt_token = UserAccessTokenFactory(user=playlist_access.user)
+        response = self.client.post(
+            f"/api/videos/{self.some_video.id}/upload-ended/",
+            {
+                "wrong_key": f"source/{self.some_video.pk}/video/4564565456",
+            },
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        self.assertEqual(response.status_code, 400)
+
+    def test_end_upload_with_forged_path(self):
+        """Playlist administrator can end an upload."""
+        playlist_access = factories.PlaylistAccessFactory(
+            playlist=self.some_video.playlist,
+            role=models.ADMINISTRATOR,
+        )
+
+        jwt_token = UserAccessTokenFactory(user=playlist_access.user)
+        response = self.client.post(
+            f"/api/videos/{self.some_video.id}/upload-ended/",
+            {
+                "file_key": f"tmp/{self.some_video.pk}/crafted/4564565456",
+            },
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        self.assertEqual(response.status_code, 400)
+
+    def test_end_upload_with_forged_stamp(self):
+        """Playlist administrator can end an upload."""
+        playlist_access = factories.PlaylistAccessFactory(
+            playlist=self.some_video.playlist,
+            role=models.ADMINISTRATOR,
+        )
+
+        jwt_token = UserAccessTokenFactory(user=playlist_access.user)
+        response = self.client.post(
+            f"/api/videos/{self.some_video.id}/upload-ended/",
+            {
+                "wrong_key": f"tmp/{self.some_video.pk}/video/crafted_stamp",
+            },
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        self.assertEqual(response.status_code, 400)
```

Comment:
Test using LTI jwt token are missing. Take a look in the `test_create` for exemple, they are at the top of the test suite